src/Application/Questions/Queries/QuestionsQuery/QuestionsQuery.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Codidact.Core.Application.Common.Contracts;
+using Codidact.Core.Application.Common.Interfaces;
+using Codidact.Core.Domain.Entities;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+
+namespace Codidact.Core.Application.Questions.Queries.QuestionsQuery
+{
+    /// <summary>
+    /// Returns a sets of questions based on the request
+    /// </summary>
+    public class QuestionsQuery : IRequestHandler<QuestionsQueryRequest, QuestionsQueryResult>
+    {
+        private readonly IApplicationDbContext _context;
+        private readonly ILogger<QuestionsQuery> _logger;
+
+        public QuestionsQuery(IApplicationDbContext context, ILogger<QuestionsQuery> logger)
+        {
+            _context = context;
+            _logger = logger;
+        }
+
+        public async Task<QuestionsQueryResult> Handle(QuestionsQueryRequest request)
+        {
+            _logger.LogInformation($"{DateTime.UtcNow.ToString("g")} - Starting to handle request for Questions List");
```

Comment:
Likewise, use UTC-ISO8601 rather than locale-specific. Server logs will need to be readable in more than one country :)